File: sdk/dotnet/src/Cordum.Cap/Subjects.cs
namespace Cordum.Cap;

/// <summary>
/// NATS subject constants and protocol defaults for the Cordum Agent Protocol.
/// </summary>
public static class Subjects
{
    public const string Submit = "sys.job.submit";
    public const string Result = "sys.job.result";
    public const string Heartbeat = "sys.heartbeat";
    public const string Alert = "sys.alert";
    public const string Progress = "sys.job.progress";
    public const string Cancel = "sys.job.cancel";
    public const string Dlq = "sys.job.dlq";
    public const string WorkflowEvent = "sys.workflow.event";
    public const string Handshake = "sys.handshake";

    public const int DefaultProtocolVersion = 1;
    public static readonly TimeSpan DefaultHeartbeatInterval = TimeSpan.FromSeconds(5);
}

File: sdk/java/src/main/java/io/cordum/cap/Subjects.java
package io.cordum.cap;

public final class Subjects {
    public static final String SUBMIT = "sys.job.submit";
    public static final String RESULT = "sys.job.result";
    public static final String HEARTBEAT = "sys.heartbeat";
    public static final String ALERT = "sys.alert";
    public static final String PROGRESS = "sys.job.progress";
    public static final String CANCEL = "sys.job.cancel";
    public static final String DLQ = "sys.job.dlq";
    public static final String WORKFLOW_EVENT = "sys.workflow.event";
    public static final String HANDSHAKE = "sys.handshake";

    public static final int DEFAULT_PROTOCOL_VERSION = 1;
    public static final long DEFAULT_HEARTBEAT_INTERVAL_MS = 5000;

    private Subjects() {}
}
